modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/restclients/SCIM2RestClient.java

```diff
@@ -880,6 +880,128 @@ public int attemptUserUpdate(PatchOperationRequestObject patchUserInfo, String u
+                errorObj.put("rawResponse", responseString);
+                errorObj.put("statusCode", statusCode);
+                return errorObj;
+            }
+        } catch (Exception e) {
+            JSONObject errorObj = new JSONObject();
+            errorObj.put("error", "Exception: " + (e.getMessage() != null ? e.getMessage() : e.getClass().getName()));
+            return errorObj;
+        }
+    }
+
+    /**
+     * Create a user with the provided bearer token and return the full response as a JSONObject.
+     * This method does not assert the response status code, allowing the caller to handle different scenarios.
+     *
+     * @param appRegisteredUserInfo UserObject containing user details.
+     * @param token                 Bearer token for authentication.
+     * @return JSONObject containing the full response, including status code and any error messages.
+     */
+    public JSONObject createUserWithBearerTokenAndReturnResponse(UserObject appRegisteredUserInfo, String token) {
+        String jsonRequest = toJSONString(appRegisteredUserInfo);
+        if (appRegisteredUserInfo.getScimSchemaExtensionEnterprise() != null) {
+            jsonRequest = jsonRequest.replace(SCIM_SCHEMA_EXTENSION_ENTERPRISE, USER_ENTERPRISE_SCHEMA);
+        }
+        if (appRegisteredUserInfo.getScimSchemaExtensionSystem() != null) {
+            jsonRequest = jsonRequest.replace(SCIM_SCHEMA_EXTENSION_SYSTEM, USER_SYSTEM_SCHEMA);
+        }
+
+        try (CloseableHttpResponse response = getResponseOfHttpPost(getUsersPath(), jsonRequest,
+                getHeadersWithBearerToken(token))) {
+            int statusCode = response.getStatusLine().getStatusCode();
+            String responseString = response.getEntity() != null ? EntityUtils.toString(response.getEntity()) : "";
+            if (responseString == null || responseString.trim().isEmpty()) {
+                JSONObject result = new JSONObject();
+                result.put("status", statusCode);
+                result.put("message", "Empty response body");
+                return result;
+            }
+            try {
+                return getJSONObject(responseString);
+            } catch (org.json.simple.parser.ParseException pe) {
+                JSONObject errorObj = new JSONObject();
+                errorObj.put("error", "ParseException: " + pe.getMessage());
+                errorObj.put("rawResponse", responseString);
+                errorObj.put("statusCode", statusCode);
+                return errorObj;
+            }
+        } catch (Exception e) {
+            JSONObject errorObj = new JSONObject();
+            errorObj.put("error", "Exception: " + (e.getMessage() != null ? e.getMessage() : e.getClass().getName()));
+            return errorObj;
+        }
+    }
+
+    /**
+     * Get the ID of a user by username.
+     *
+     * @param s Username of the user.
+     * @return ID of the user.
+     */
+    public String getUserId(String s) {
```

Comment:
```suggestion
    public String getUserIdFromUsername(String username) {
```

Also move this method to top. Near getUserId() method